cw-orch/src/daemon/queriers/node.rs

```diff
@@ -184,4 +192,78 @@ impl Node {
         // return error if tx not found by now
         Err(DaemonError::TXNotFound(hash, retries))
     }
+
+    /// Find TX by events
+    pub async fn find_tx_by_events(
+        &self,
+        events: Vec<String>,
+        page: Option<u64>,
+        order_by: Option<OrderBy>,
+    ) -> Result<Vec<CosmTxResponse>, DaemonError> {
+        self.find_tx_by_events_with_retries(events, page, order_by, false, MAX_TX_QUERY_RETRIES)
+            .await
+    }
+
+    /// Find Tx by events and waits for until there is a non-empty response
+    pub async fn find_some_tx_by_events(
+        &self,
+        events: Vec<String>,
+        page: Option<u64>,
+        order_by: Option<OrderBy>,
+    ) -> Result<Vec<CosmTxResponse>, DaemonError> {
+        self.find_tx_by_events_with_retries(events, page, order_by, true, MAX_TX_QUERY_RETRIES)
+            .await
+    }
+
+    /// Find TX by events with  :
+    /// 1. Specify if an empty tx object is a valid response
+    /// 2. Specify a given amount of retries
+    pub async fn find_tx_by_events_with_retries(
+        &self,
+        events: Vec<String>,
+        page: Option<u64>,
+        order_by: Option<OrderBy>,
+        retry_on_empty: bool,
+        retries: usize,
+    ) -> Result<Vec<CosmTxResponse>, DaemonError> {
+        let mut client =
+            cosmos_modules::tx::service_client::ServiceClient::new(self.channel.clone());
+
+        let request = cosmos_modules::tx::GetTxsEventRequest {
+            events: events.clone(),
+            page: page.unwrap_or(0),
+            limit: 100,
+            pagination: None, // This is not used, so good.
+            order_by: order_by.unwrap_or(OrderBy::Desc).into(),
+        };
+
+        for _ in 0..retries {
+            match client.get_txs_event(request.clone()).await {
+                Ok(tx) => {
+                    let resp = tx.into_inner().tx_responses;
+                    if retry_on_empty && resp.is_empty() {
```

Comment:
if resp is empty, what does that mean? Does that mean the tx is still pending? If so, we should rename above functions because it's unclear what this does rn.

Reply:
It means that there is not event corresponding to the array you passed. In some cases, it's good, we got the results, in some cases we need to wait until there is a transaction that fits the events.